src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -86,24 +100,40 @@ public static CollectorManager createHybridCollectorManager(final SearchContext
                 isSingleShard,
                 trackTotalHitsUpTo,
                 searchContext.sort(),
-                filteringWeight
+                filteringWeight,
+                searchContext.searchAfter()
             )
             : new HybridCollectorNonConcurrentManager(
                 numDocs,
                 new HitsThresholdChecker(Math.max(numDocs, searchContext.trackTotalHitsUpTo())),
                 isSingleShard,
                 trackTotalHitsUpTo,
                 searchContext.sort(),
-                filteringWeight
+                filteringWeight,
+                searchContext.searchAfter()
             );
     }
 
     @Override
     public Collector newCollector() {
-        Collector hybridcollector = new HybridTopScoreDocCollector(numHits, hitsThresholdChecker);
-        // Check if filterWeight is present. If it is present then return wrap Hybrid collector object underneath the FilteredCollector
+        if (sortAndFormats == null) {
+            // Hybrid Query
+            // Check if filterWeight is present. If it is present then return wrap Hybrid collector object underneath the FilteredCollector
+            // object and return it.
+            Collector hybridcollector = new HybridTopScoreDocCollector(numHits, hitsThresholdChecker);
+            return Objects.nonNull(filterWeight) ? new FilteredCollector(hybridcollector, filterWeight) : hybridcollector;
+        }
+        // Hybrid Query with Sorting Applied
+        Collector hybridSortcollector;
+        if (after == null) {
+            hybridSortcollector = new SimpleFieldCollector(numHits, hitsThresholdChecker, sortAndFormats.sort);
+        } else {
+            validateSearchAfterFieldAndSortFormats();
+            hybridSortcollector = new PagingFieldCollector(numHits, hitsThresholdChecker, sortAndFormats.sort, after);
+        }
+        // Check if filterWeight is present. If it is present then return wrap Hybrid Sort collector object underneath the FilteredCollector
```

Comment:
you can do whole this method simpler. Write one private method that will give you a low level collector that is either hybrid topscore or one of sorting specific collectors. Then with that collector do the old logic. It should be possible as all collectors are "isa" Collector interface